libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/ScreenLock/ScreenLockManagerInternal.swift

```diff
@@ -0,0 +1,284 @@
+            SFSDKCoreLogger.e(ScreenLockManagerInternal.self, message: "Failed to store global mobile policy.")
+        }
+    }
+    
+    func removeGlobalPolicy() {
+        let globalResult = KeychainHelper.remove(service: kScreenLockIdentifier, account: "global")
+        if globalResult.success {
+            SFSDKCoreLogger.i(ScreenLockManagerInternal.self, message: "Global mobile policy removed.")
+        } else {
+            SFSDKCoreLogger.e(ScreenLockManagerInternal.self, message: "Failed to remove global mobile policy.")
+        }
+    }
+    
+    func unlock() {
+        // Send flow will begin notification
+        SFSDKCoreLogger.d(ScreenLockManagerInternal.self, message: "Sending screen lock flow completed notification")
+        NotificationCenter.default.post(name: Notification.Name(rawValue: kSFScreenLockFlowCompleted), object: nil)
+        
+        SFSDKWindowManager.shared().screenLockWindow().dismissWindow()
+        unlockPostProcessing()
+    }
+    
+    private func unlockPostProcessing() {
+        if callbackBlock != nil {
+            let callbackBlockCopy = callbackBlock!
+            callbackBlock = nil
+            callbackBlockCopy()
+        } else {
+            SFSDKCoreLogger.e(ScreenLockManagerInternal.self, message: "callbackBlock is nil.")
+        }
+    }
+
+    func lock() {
+        if !Thread.isMainThread {
+            DispatchQueue.main.async {
+                self.lock()
+            }
+        }
+        
+        if SFSDKWindowManager.shared().snapshotWindow(nil).isEnabled() {
+            SFSDKWindowManager.shared().snapshotWindow(nil).dismissWindow()
+        }
+        
+        // Send flow will begin notification
+        SFSDKCoreLogger.d(ScreenLockManagerInternal.self, message: "Sending Screen Lock flow will begin notification")
+        NotificationCenter.default.post(name: Notification.Name(rawValue: kSFScreenLockFlowWillBegin), object: nil)
+        
+        // Launch Screen Lock
+        let screenLockViewController = UIHostingController(rootView: ScreenLockUIView())
+        screenLockViewController.modalPresentationStyle = .fullScreen
+        SFSDKWindowManager.shared().screenLockWindow().presentWindow(animated: false) {
+            SFSDKWindowManager.shared().screenLockWindow().viewController?.present(screenLockViewController, animated: false, completion: nil)
+        }
+    }
+    
+    @objc func checkForPolicy(userId: String) -> Bool {
+        let result = KeychainHelper.read(service: kScreenLockIdentifier, account: userId)
+        if let resultData = result.data, result.success {
+            do {
+                return try JSONDecoder().decode(MobilePolicy.self, from: resultData).hasPolicy
+            } catch {
+                SFSDKCoreLogger.e(ScreenLockManagerInternal.self, message: "Failed to read Mobile policy for user.")
+            }
+        }
+        
+        return false
+    }
+
+    private struct MobilePolicy: Encodable, Decodable {
```

Comment:
Would it make sense to have a single method that reads from the keychain and returns the MobilePolicy object.
In all the other places, we would call that method and then access the field of interest.
That way we wouldn't have the following repeated multiple times:
```swift
let result = KeychainHelper.read(service: kScreenLockIdentifier, account: userId)
if let resultData = result.data, result.success {
  do {
    try JSONDecoder().decode(MobilePolicy.self, from: resultData)
  } catch {
      SFSDKCoreLogger.e(ScreenLockManagerInternal.self, message: "Failed to read Mobile policy for user.")
  }
}
```